src/NuGet.Clients/NuGet.PackageManagement.UI/Resources/Resources.xaml.cs

```diff
@@ -13,7 +17,10 @@ public SharedResources()
         private void PackageIconImage_ImageFailed(object sender, ExceptionRoutedEventArgs e)
         {
             var image = sender as Image;
-            image.Source = Images.DefaultPackageIcon;
+
+            e.Handled = true; // don't repropagate the event
+            var binding = BindingOperations.GetBindingExpression(image, Image.SourceProperty);
+            binding.UpdateTarget();
```

Comment:
As per this [doc](https://docs.microsoft.com/en-us/dotnet/api/system.windows.controls.image.imagefailed?view=netframework-4.8), `ImageFailed` event occurs when there is a failure in the image. Do we know why this event is getting fired even though image loaded properly when rendered for the first time?

Reply:
Because the `Image` element is being recycled by the VirtualizingStackPanel. So, when the event arrive a the `Image` control event handler, the control is processing an event whose source image does not correspond to the current image that is loaded.